VideoWeb/VideoWeb/ClientApp/src/app/services/connection-status.service.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { MockLogger } from '../testing/mocks/MockLogger';
+import { Logger } from './logging/logger-base';
+import { ConnectionStatusService } from '../services/connection-status.service';
+
+describe('ConnectionStatusService', () => {
+    let service: ConnectionStatusService;
+    const logger: Logger = new MockLogger();
+
+    beforeAll(() => {
+        service = new ConnectionStatusService(logger);
+    });
+
+    beforeEach(function () {
+        jasmine.clock().install();
+    });
+
+    afterEach(() => {
+        jasmine.clock().uninstall();
+    });
+
+    it('should send request to check internet connection', () => {
+        // Arrange
+        spyOn(XMLHttpRequest.prototype, 'open').and.callFake((method: string, url: string) => {});
+        spyOn(XMLHttpRequest.prototype, 'send');
+
+        // Act
+        service.checkNow();
+
+        // Assert
+        expect(XMLHttpRequest.prototype.open).toHaveBeenCalled();
+        expect(XMLHttpRequest.prototype.send).toHaveBeenCalled();
+    });
+
+    it('should stop timer', () => {
+        // Arrange
+        spyOn(XMLHttpRequest.prototype, 'open').and.callFake((method: string, url: string) => {});
+        spyOn(XMLHttpRequest.prototype, 'send');
+
+        // Act
+        service.start();
+        jasmine.clock().tick(5001);
+        service.stopTimer();
+        jasmine.clock().tick(60000);
+
+        // Assert
+        expect(XMLHttpRequest.prototype.open).toHaveBeenCalledTimes(2);
+        expect(XMLHttpRequest.prototype.send).toHaveBeenCalledTimes(2);
+    });
+
+    it('should stop timer if not started', () => {
+        expect(() => service.stopTimer()).not.toThrow();
+    });
+
+    it('should setup interval and fire intial check and again every 5 seconds', () => {
```

Comment:
Not a big fan of a test that intentionally waits for 5+ seconds. The current test suit runs in half that time.

Reply:
Would you consider the ability to change the interval and then have then use the much smaller time period?

Reply:
not waiting :) but yeah i wouldnt be either :P